Quarkus throws NPE when trying to match unknown config files

### Describe the bug

https://github.com/quarkusio/quarkus/pull/37520 we get NPE when building our application with config sources.

### Expected behavior

No NPE.

### Actual behavior

Build fails with following exception:

```
ERROR] Failed to execute goal io.quarkus:quarkus-maven-plugin:999-SNAPSHOT:build (build) on project properties: Failed to build quarkus application: io.quarkus.builder.BuildException: Build failure: Build failed due to errors
[ERROR] 	[error]: Build step io.quarkus.deployment.steps.ConfigGenerationBuildStep#unknownConfigFiles threw an exception: java.lang.NullPointerException: Cannot invoke "java.lang.CharSequence.toString()" because "s" is null
[ERROR] 	at java.base/java.lang.String.contains(String.java:2977)
[ERROR] 	at io.quarkus.runtime.configuration.ConfigDiagnostic.unknownConfigFiles(ConfigDiagnostic.java:231)
[ERROR] 	at io.quarkus.deployment.steps.ConfigGenerationBuildStep.unknownConfigFiles(ConfigGenerationBuildStep.java:439)
[ERROR] 	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
[ERROR] 	at java.base/java.lang.reflect.Method.invoke(Method.java:580)
[ERROR] 	at io.quarkus.deployment.ExtensionLoader$3.execute(ExtensionLoader.java:849)
[ERROR] 	at io.quarkus.builder.BuildContext.run(BuildContext.java:256)
[ERROR] 	at org.jboss.threads.ContextHandler$1.runWith(ContextHandler.java:18)
[ERROR] 	at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2513)
[ERROR] 	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1538)
[ERROR] 	at java.base/java.lang.Thread.run(Thread.java:1583)
[ERROR] 	at org.jboss.threads.JBossThread.run(JBossThread.java:501)
[ERROR] -> [Help 1]
```

### How to Reproduce?

Steps to reproduce the issue:

1. `git clone git@github.com:quarkus-qe/quarkus-test-suite.git`
2. `quarkus-test-suite/properties`
3. `mvn clean verify`

### Output of `uname -a` or `ver`

Fedora 38

### Output of `java -version`

openjdk version "21.0.1" 2023-10-17 LTS

### Quarkus version or git rev

999-SNAPSHOT

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

Apache Maven 3.9.3

### Additional information

_No response_

--- a/core/runtime/src/main/java/io/quarkus/runtime/configuration/ConfigDiagnostic.java
+++ b/core/runtime/src/main/java/io/quarkus/runtime/configuration/ConfigDiagnostic.java
@@ -184,7 +184,7 @@ public final class ConfigDiagnostic {
         Set<String> configFiles = new HashSet<>();
         try (DirectoryStream<Path> candidates = Files.newDirectoryStream(configFilesLocation, CONFIG_FILES_FILTER)) {
             for (Path candidate : candidates) {
-                configFiles.add(candidate.toUri().getPath());
+                configFiles.add(candidate.toUri().toURL().toString());
             }
         }
         return configFiles;

